src/optim/FishLeg/fishleg.py

```diff
@@ -240,92 +174,108 @@ def update_aux(self) -> None:
         method = group["method"]
         method_kwargs = group["method_kwargs"]
 
-        pred_y = self.model(data_x)
-
         with torch.no_grad():
+            pred_y = self.model(data_x)
             samples_y = self.likelihood.draw(pred_y)
 
-        loss = self.likelihood.nll(pred_y, samples_y)
-
-        grads = torch.autograd.grad(
-            outputs=loss,
-            inputs=group["params"],
-            allow_unused=True,
-        )
-
-        u2 = 0.0
-        for group in self.param_groups:
-            for params in group["params"]:
-                u2 = u2 + torch.sum(params.grad * params.grad)
-        u_norm = torch.sqrt(u2)
+        g2 = 0.0
+        for g in group["params"]:
+            g2 += torch.sum(g**2)
```

Comment:
isnt the name g confusing? as we are not using gradients here?

Reply:
this should be params.grad, otherwise it will take the parameter value instead of the gradient

Reply:
grad.data

Reply:
248 layer_grads